etna/collections/models.py

```diff
@@ -145,12 +150,38 @@ def related_articles(self):
         from etna.articles.models import ArticlePage
 
         return (
-            ArticlePage.objects.filter(topic=self)
-            .live()
-            .select_related("teaser_image")
-            .order_by("title")[:3]
+            ArticlePage.objects.live()
+            .public()
+            .filter(id__in=self.related_page_ids)
+            .order_by("-first_published_at")
+            .exclude(title=self.featured_article)
+        )
+
+    @cached_property
+    def related_record_articles(self):
+        from etna.articles.models import RecordArticlePage
+
+        return (
+            RecordArticlePage.objects.live()
+            .public()
+            .filter(id__in=self.related_page_ids)
+            .order_by("-first_published_at")
+            .exclude(title=self.featured_record_article)
```

Comment:
Thought: Is this right? if the featured_record_article is `None` (as it can be left blank), then you're filtering a QS via a reference (PK). I think you'll find this produces an unintended results. You're better off with `exclude(pk=self.featured_record_article)`.

Have a look at this SO post here; https://stackoverflow.com/questions/2165865/django-queries-id-vs-pk it'll explain a little more about the difference between PK V.S. ID. I would suggest using PK going forward.